### PR TITLE
Feature/validation updates

### PR DESCRIPTION
**Pull request by bramirez96**
# Description

This pull is part of a larger pass at the server that intends to make router setups and documentation more consistent. This includes creating middleware to validate parent and child data as well as rewriting certain blocks of code to match the style of the more recent routers. Tests and documentation will need to be updated for parents and children to account for the new possibility of a 400 error being passed. Also included in this update will be a fix to the child data model so that, rather than pulling foreign keys for their avatar and grade level, it uses joins to pull the relevant information.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [x] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Review by SandraCoburn — APPROVED** [review]
Great updates. Tests look good.

**Review by Nick-Ohman — APPROVED** [review]
Great tests, nice job!

File: __tests__/routes/avatars.test.js [hunk 1]
```diff
@@ -35,7 +42,7 @@ describe('avatar router endpoints', () => {
     });
 
     it('should restrict the addition of redundant avatars', async () => {
```
Comment (Schrese) [inline __tests__/routes/avatars.test.js:44]: A great thing to check for, I didn't even think about that

**Review by Schrese — APPROVED** [review]
This is looking great and the documentation is very concise!

**Pull request merged**